src/plugins/data_source/config.ts

```diff
@@ -31,6 +31,15 @@ export const configSchema = schema.object({
       defaultValue: new Array(32).fill(0),
     }),
   }),
+  ssl: schema.object({
+    verificationMode: schema.oneOf(
+      [schema.literal('none'), schema.literal('certificate'), schema.literal('full')],
+      { defaultValue: 'full' }
+    ),
+    certificateAuthorities: schema.maybe(
+      schema.oneOf([schema.string(), schema.arrayOf(schema.string(), { minSize: 1 })])
+    ),
```

Comment:
The `minSize` unnecessarily complicates this; why should we not consider a blank array as undefined?

Reply:
This is the same as the client for [default opensearch](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/opensearch/opensearch_config.ts#L103-L105). The ways its implemented in this PR handles empty as you describe so it can be removed but I had this here to keep it consistent with the other place its defined for default opensearch.

Reply:
That is great. i think we can progressively enhance these. I am glad to hear that the code can handle that case.